cms/djangoapps/contentstore/views/course.py

```diff
@@ -1460,32 +1460,108 @@ def _get_usage_info(store, course, split_tests):
 
             validation_summary = split_test.general_validation_message()
             usage_info[split_test.user_partition_id].append({
-                'label': '{} / {}'.format(unit.display_name, split_test.display_name),
+                'label': u"{} / {}".format(unit.display_name, split_test.display_name),
                 'url': unit_url,
                 'validation': validation_summary.to_json() if validation_summary else None,
             })
         return usage_info
 
     @staticmethod
+    def get_content_groups_usage_info(store, course):
+        """
+        Get usage information for content Group Configurations.
+        """
+        modules = store.get_items(course.id, settings={'group_access': {'$exists': True}})
+
+        return GroupConfiguration._get_content_groups_usage_info(course, modules)
+
+    @staticmethod
+    def _get_content_groups_usage_info(course, modules):
+        """
+        Returns all units names and their urls.
+
+        Returns:
+        {'group_id':
+            [
+                {
+                    'label': 'Unit 1 / Problem 1',
+                    'url': 'url_to_unit_1'
+                },
+                {
+                    'label': 'Unit 2 / Problem 2',
+                    'url': 'url_to_unit_2'
+                }
+            ],
+        }
+        """
+        usage_info = {}
+        for module in modules:
+            if hasattr(module, 'group_access') and module.group_access:
+                (__, group_ids), = module.group_access.items()
+                for group_id in group_ids:
+                    if group_id not in usage_info:
+                        usage_info[group_id] = []
+
+                    module_parent = module.get_parent()
+
+                    if not module_parent:
+                        log.warning("Parent module of content_groups not found: %s", module.location)
+                        continue
+
+                    unit_url = reverse_usage_url(
```

Comment:
It looks like you can share the logic for constructing usage info for an xblock between this and ``_get_usage_info``.